app/stac_api/models.py

```diff
@@ -120,46 +118,44 @@ class Collection(models.Model):
     # furthermore GeoDjango and its functionality will be used for that.
     # TODO: overwrite items save() function accordingly
     # suggestions of fields to be auto-populated:
-    extent = ArrayField(models.FloatField(), blank=True)  # [Float], auto-populated from items
-
+    extent = models.JSONField(blank=True, null=True)  # [Float], auto-populated from items
     collection_name = models.CharField(unique=True, max_length=255)  # string
     # collection_name is what is simply only called "id" in here:
     # http://ltboc.infra.bgdi.ch/static/products/data.geo.admin.ch/apitransactional.html#operation/createCollection
     item_type = models.CharField(default="Feature", max_length=20)  # string
-
     keywords = models.ManyToManyField(Keyword)
     license = models.CharField(max_length=30)  # string
     providers = models.ManyToManyField(Provider)
-
     stac_extension = ArrayField(
         models.CharField(max_length=255), default=get_default_stac_extensions, editable=False
     )
     stac_version = models.CharField(max_length=10)  # string
 
     # "summaries" values will be updated on every update of an asset inside the
     # collection
-    summaries_eo_gsd = ArrayField(models.FloatField(), blank=True, null=True)
-    summaries_proj = ArrayField(models.IntegerField(), blank=True, null=True)
+    summaries = models.JSONField(blank=True, null=True)
+    #summaries_eo_gsd = ArrayField(models.FloatField(), blank=True, null=True)
+    #summaries_proj = ArrayField(models.IntegerField(), blank=True, null=True)
     # after discussion with Chris and Tobias: geoadmin_variant will be an
     # array field of CharFields. Simple validation is done (e.g. no "Sonderzeichen"
     # in array)
     # geoadmin_variant will be also auto-populated on every
     # update of an asset
-    geoadmin_variant = ArrayField(models.CharField(max_length=15), blank=True, null=True)
-
+    #geoadmin_variant = ArrayField(models.CharField(max_length=15), blank=True, null=True)
```

Comment:
Clean up

Reply:
as for all of your "KeyError" comments: I will add handling, but as these fields are only for internal write access, the requested keys and indices should always work. But yes, if something is wrong with the internal writing, this would cause errors. I'll add that soon